src/sparseml/pytorch/sparsification/pruning/modifier_pruning_base.py

```diff
@@ -319,7 +325,8 @@ def initialize(
 
         self.initialize_extras(module)
 
-        self.check_mask_update(module, epoch, steps_per_epoch=1, **kwargs)
+        if self.is_oneshot(epoch, steps_per_epoch=1):
```

Comment:
so this just checks if epoch is inf then? out of curiosity what's the point of steps_per_epoch here?

Reply:
To the best of my knowledge, `steps_per_epoch=1` is convention used in the codebase for these "one-shot" use-cases. That's why the `check_mask_update` in the line below is also called with `steps_per_epoch=1` (when called like this from the `initialize` method).

`is_oneshot` method in this specific use-case will check if epoch is `inf`, but in general use-case it makes it explicit what we consider as one-shot throughout the codebase. For example, gradient-based pruners need to check somehow if they are called in one-shot or gradual setup, and (hopefully) this will be a unified way to do that by calling: `if self.is_oneshot(...)`.

Reply:
Gotcha, thanks for the details!